lib/api/CNdJsonInputParser.cc

```diff
@@ -111,24 +111,44 @@ bool CNdJsonInputParser::readStreamIntoVecs(const TVecReaderFunc& readerFunc,
     return true;
 }
 
-bool CNdJsonInputParser::parseDocument(char* begin, rapidjson::Document& document) {
-    // Parse JSON string using Rapidjson
-    if (document.ParseInsitu<rapidjson::kParseStopWhenDoneFlag>(begin).HasParseError()) {
-        LOG_ERROR(<< "JSON parse error: " << document.GetParseError());
+bool CNdJsonInputParser::parseDocument(char* begin, json::value& document) {
```

Comment:
You can easily add a `length` argument to this method, as `CNdInputParser::parseLine` returns both a start and a length. Only the start is currently passed, because that's all RapidJSON used. But by passing the length as well you can avoid the extra buffer and loop in this method. All of that can just be replaced with:

```
    p.write_some(begin, sizeof(length), ec)
```

where `begin` is the existing argument and `length` is the new one.

Reply:
Got it. Thanks!